### PR TITLE
refactor(mock-server): use workspace-store for $ref resolution

### PR DESCRIPTION
**Pull request by hanspagel**
Switches the mock server from `@scalar/openapi-parser` to `@scalar/workspace-store` + `@scalar/json-magic` for OpenAPI reference handling.

Instead of eagerly dereferencing the whole document up front, it now keeps `$ref`s in place (via the json-magic magic proxy) and resolves them lazily with `getResolvedRef` where consumers actually read responses, schemas, and security schemes. 

Example generation deep-resolves the relevant schema first (`getResolvedRefDeep`), matching how workspace-store does it internally. Without it, a ref nested under another ref (like galaxy's `Planet.creator → User`) renders as `null`. The `/openapi.{json,yaml}` endpoints now use `@scalar/json-magic`'s `normalize` and the `yaml` package.

No behavior change, all 193 mock-server tests pass.

---

> [!NOTE]
> **Medium Risk**
> Changes how every mock route reads OpenAPI (responses, auth, OAuth setup); behavior is intended to stay the same but ref edge cases and example generation paths shifted.
> 
> **Overview**
> Replaces eager full-document dereferencing (`@scalar/openapi-parser`) with **lazy** `$ref` handling: after bundle + upgrade, the spec is wrapped in `@scalar/json-magic`'s magic proxy and resolved at read time via `getResolvedRef` / `getResolvedRefDeep` from `@scalar/workspace-store`.
> 
> Routing, mocks, auth, and seeding now resolve refs where they read the spec—path items, responses, headers, schemas (with deep resolve for examples), and security schemes—including **`mergeSiblingReferences`** for `x-seed` next to a `$ref`. Broken refs are skipped instead of failing document processing; new tests cover startup and responses with missing components.
> 
> **`/openapi.json`** and **`/openapi.yaml`** serialize with `@scalar/json-magic` **`normalize`** and the **`yaml`** package instead of openapi-parser helpers. Dependency swap drops **`@scalar/openapi-parser`**, adds **`yaml`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f1a5a149681356a4aab883787abf9e943adb4f7d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by marclave — APPROVED** [review]

File: packages/mock-server/src/utils/process-openapi-document.ts [hunk 1]
```diff
@@ -63,26 +65,7 @@ export async function processOpenApiDocument(
...
-
-  // Ensure the schema is a valid OpenAPI 3.1 document
-  if (typeof schema !== 'object') {
-    throw new Error('Dereferenced schema is invalid: expected an object')
-  }
-
-  return schema as OpenAPIV3_1.Document
+  // Wrap the document in a magic proxy so internal references resolve lazily via `$ref-value`.
+  // External references were already pulled inline by `bundle` above, so only local `$ref`s remain.
+  return createMagicProxy(upgraded) as OpenAPIV3_1.Document
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/mock-server/src/utils/process-openapi-document.ts:70]: This one is intentional. The whole point of the change is to stop eagerly dereferencing, so we no longer walk every `$ref` at startup. Re-adding a validation pass would mean traversing the full document up front again, which defeats the lazy resolution.

To make the runtime side safe, the read sites now degrade gracefully instead of crashing on an unresolvable `$ref` (security schemes, response headers, path items), covered by `test/unresolved-references.test.ts`. A broken ref skips that piece rather than taking down the server — which for a mock server feels better than failing fast anyway.

**Review by amritk — APPROVED** [review]
